### PR TITLE
Fix linearized bondi sachs

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Split from #2609 -- this is just the part that fixes up the `LinearizedBondiSachs` solution.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### Dependencies

- [x] #2609

### CODE REVIEW

File: src/Evolution/Systems/Cce/Initialize/RegisterInitializeJWithCharm.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Evolution/Systems/Cce/AnalyticSolutions/LinearizedBondiSachs.hpp"
+#include "Evolution/Systems/Cce/Initialize/InitializeJ.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+
+namespace Cce {
```
Comment (nilsdeppe) [inline src/Evolution/Systems/Cce/Initialize/RegisterInitializeJWithCharm.hpp:11]: `/// \cond` around forward decl

**Review by nilsdeppe — COMMENTED** [review]
Looks good, feel free to squash things immediately

**Comment by moxcodes** [comment]
fixup squashed in (and rebased so no longer dependent)
Cheers!

**Review by nilsdeppe — DISMISSED** [review]

File: src/Evolution/Systems/Cce/AnalyticSolutions/LinearizedBondiSachs.cpp [hunk 2]
```diff
@@ -12,99 +12,197 @@
...
+                              z_33_harmonic * real(time_factor * l_3_factor);
+}
+}  // namespace LinearizedBondiSachs_detail
+
+LinearizedBondiSachs::LinearizedBondiSachs(
+    const std::vector<std::complex<double>>& mode_constants,
+    const double extraction_radius, const double frequency) noexcept
+    : SphericalMetricData{extraction_radius}, frequency_{frequency} {
+  auto mode_constant_copy = mode_constants;
+  mode_constant_copy.resize(2, std::complex<double>(0.0, 0.0));
```
Comment (nilsvu) [inline src/Evolution/Systems/Cce/AnalyticSolutions/LinearizedBondiSachs.cpp]: What does this resizing do? Is the intention that the user may specify only one mode, and the other defaults to zero? In that case I suggest you just remove the default behavior, which means you can switch the `std::vector` to a `std::array`. If I misunderstood this, I suggest to add a comment what's going on :)

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Comment by moxcodes** [comment]
squashed in the change to std::array input option.
Cheers!

**Review by nilsdeppe — APPROVED** [review]

**Review by nilsvu — APPROVED** [review]

**Pull request merged**